logger/honeycomb.go

```diff
@@ -66,6 +69,11 @@ func (h *HoneycombLogger) Start() error {
 		}
 	}
 
+	err = h.maybeStartSampler(loggerConfig)
```

Comment:
Not my favorite: passing the parsed config into the receiver func, but parsing the logger config _again_ feels worse.